scripts/backup/backup

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+UMBREL_ROOT="$(readlink -f $(dirname "${BASH_SOURCE[0]}")/../..)"
+BACKUP_FOLDER="backup"
+BACKUP_ROOT="${UMBREL_ROOT}/${BACKUP_FOLDER}"
+BACKUP_FILE="${UMBREL_ROOT}/backup.tar.gz.pgp"
+
+check_dependencies () {
+  for cmd in "$@"; do
+    if ! command -v "$cmd" >/dev/null 2>&1; then
+      echo "This script requires \"${cmd}\" to be installed"
+      exit 1
+    fi
+  done
+}
+
+check_dependencies openssl tar gpg shuf curl
+
+# Deterministically derives 128 bits of cryptographically secure entropy
+derive_entropy () {
+  identifier="${1}"
+  umbrel_seed=$(cat "${UMBREL_ROOT}/db/umbrel-seed/seed") || true
+
+  # TODO: Check this correctly handles when seed file doesn't exist!
+  if [[ -z "$umbrel_seed" ]] || [[ -z "$identifier" ]]; then
+    echo "Missing derivation parameter, this is unsafe, exiting."
+    rm -f "${UMBREL_ROOT}/statuses/backup-in-progress"
+    exit 1
+  fi
+
+  # We need `sed 's/^.* //'` to trim the "(stdin)= " prefix from some versions of openssl
+  echo -n "${identifier}" | openssl dgst -sha256 -hmac "${umbrel_seed}" | sed 's/^.* //'
+}
+
+# Make sure an update is not in progres
+if [[ -f "${UMBREL_ROOT}/statuses/backup-in-progress" ]]; then
+    echo "A backup is already in progress. Exiting now."
+    exit 1
+fi
+
+echo "Creating lock..."
+touch "${UMBREL_ROOT}/statuses/backup-in-progress"
+
+[[ -f "${UMBREL_ROOT}/.env" ]] && source "${UMBREL_ROOT}/.env"
+BITCOIN_NETWORK=${BITCOIN_NETWORK:-mainnet}
+
+[[ -d "${BACKUP_ROOT}" ]] && rm -rf "${BACKUP_ROOT}"
+
+echo "Deriving keys..."
+
+backup_id=$(derive_entropy "umbrel_backup_id")
+encryption_key=$(derive_entropy "umbrel_backup_encryption_key")
+
+echo "Creating backup..."
+
+mkdir -p "${BACKUP_ROOT}"
+
+cp --archive "${UMBREL_ROOT}/lnd/data/chain/bitcoin/${BITCOIN_NETWORK}/channel.backup" "${BACKUP_ROOT}/channel.backup"
+
+# We want to back up user settings too, however we currently store the encrypted
+# mnemonic in this file which is not safe to backup remotely.
+# Uncomment this in the future once we've ensured there's no critical data in
+# this file.
+# cp --archive "${UMBREL_ROOT}/db/user.json" "${BACKUP_ROOT}/user.json"
+
+echo "Adding random padding..."
+
+# Up to 10KB of random binary data
+padding="$(shuf -i 0-10240 -n 1)"
+dd if=/dev/urandom bs="${padding}" count=1 > "${BACKUP_ROOT}/.padding"
```

Comment:
Or `bs="$((RANDOM % 10<<10))"` since it's BASH anyway.

Reply:
`$RANDOM` is not uniformly random, it's a random integer between 0 and 32767.

I know we don't need this to be a cryptographically secure source since it's just adding some noise, but `shuf` is uniformly random, widely available, and the ranges are much more readable.